File: DGVTemplateEditor/Types/DesignColumn.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;

namespace DGVTemplateEditor.Types
{
	public class DesignColumn
	{
		private static int Count = 0;

		public DesignColumn()
		{
			Name = $"Column{Count++}";
			Header = Name;
			Width = 100;
			AutoSizeMode = DataGridViewAutoSizeColumnMode.NotSet;
			Visible = true;
			ForeColor = Color.Black;
		}

		public DesignColumn(string name, string header, int width, DataGridViewAutoSizeColumnMode autoSizeMode, bool visible) : this()
		{
			Name = name;
			Header = header;
			Width = width;
			AutoSizeMode = autoSizeMode;
			Visible = visible;
		}

		public DesignColumn(DGVTemplateColumn C) : this()
		{
			Name = C.Name;
			Header = C.Header;
			Width = C.Width;
			AutoSizeMode = C.AutoSizeMode;
			Visible = C.Visible;
			ForeColor = ColorTranslator.FromHtml(C.ForeColor);
		}

		public DGVTemplateColumn ToTemplate()
		{
			return new DGVTemplateColumn()
			{
				Name = Name,
				Header = Header,
				Width = Width,
				AutoSizeMode = AutoSizeMode,
				Visible = Visible,
				ForeColor = ColorTranslator.ToHtml(ForeColor)
			};
		}

		[Description("Имя столбца")]
		public string Name { get; set; }

		[Description("Заголовок столбца")]
		public string Header { get; set; }

		[Description("Ширина столбца")]
		public int Width { get; set; }

		[Description("Способ регулирования ширины столбца"), DefaultValue(typeof(DataGridViewAutoSizeColumnMode), "NotSet")]
		public DataGridViewAutoSizeColumnMode AutoSizeMode { get; set; }

		[Description("Видимость столбца"), DefaultValue(true)]
		public bool Visible { get; set; }

		[Description("Цвет текста"), DefaultValue(typeof(Color), "0x000000")]
		public Color ForeColor { get; set; }
	}
}

File: JANL/Types/DesignColumn.vb
Imports System.ComponentModel
Imports System.Drawing

Namespace Types

	Friend Class DesignColumn
		Inherits ObservableObject
		Private Shared Count As Integer = 0

		Public Sub New()
			Name = $"Column{Count}"
			Header = Name
			Width = 100
			AutoSizeMode = DataGridViewAutoSizeColumnMode.NotSet
			Visible = True
			ForeColor = Color.Black
			Count += 1
		End Sub

		Public Sub New(C As DGVTemplateColumn)
			Name = C.Name
			Header = C.Header
			Width = C.Width
			AutoSizeMode = C.AutoSizeMode
			Visible = C.Visible
			ForeColor = ColorTranslator.FromHtml(C.ForeColor)
		End Sub

		Public Function ToTemplate() As DGVTemplateColumn
			Return New DGVTemplateColumn() With {
				.Name = Name,
				.Header = Header,
				.Width = Width,
				.AutoSizeMode = AutoSizeMode,
				.Visible = Visible,
				.ForeColor = ColorTranslator.ToHtml(ForeColor)
			}
		End Function

		<Description("Имя столбца")>
		Public Property Name As String

		<Description("Заголовок столбца")>
		Public Property Header As String

		<Description("Ширина столбца"), DefaultValue(100)>
		Public Property Width As Integer

		<Description("Способ регулирования ширины столбца"), DefaultValue(GetType(DataGridViewAutoSizeColumnMode), "NotSet")>
		Public Property AutoSizeMode As DataGridViewAutoSizeColumnMode

		<Description("Видимость столбца"), DefaultValue(True)>
		Public Property Visible As Boolean

		<Description("Цвет текста"), DefaultValue(GetType(Color), "0x000000")>
		Public Property ForeColor As Color

		'<Description("Шрифт текста"), DefaultValue(GetType(Color), "0x000000"), Category("У")>
		'Public Property Font As Font
		Public Overrides Function ToString() As String
			Return $"{Name}({Header}, {Width})"
		End Function

	End Class

End Namespace